runtime/commonMain/src/kotlinx/serialization/internal/Enums.kt

```diff
@@ -5,13 +5,18 @@
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
+import kotlin.jvm.JvmOverloads
 import kotlin.reflect.KClass
 
-public class EnumDescriptor(override val name: String, private val choices: Array<String>) : SerialClassDescImpl(name) {
+@InternalSerializationApi
+public class EnumDescriptor @JvmOverloads constructor(
+    override val name: String,
+    private val values: Array<String> = emptyArray()
```

Comment:
`values` should not be `val`.
All required state for eq/hc is already present in the descriptor elements